### PR TITLE
Add platformSupported flag to enable daemon mode by platform.

### PR DESCRIPTION
**Pull request by calavera**
Supersedes #15162

/cc @tiborvass

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

File: daemon/daemon_unix.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// +build !windows
+// +build linux,freebsd
```
Comment (tiborvass) [inline daemon/daemon_unix.go]: what you wrote is `linux AND freebsd` it should be `OR`: `linux freebsd`

Comment (calavera) [inline daemon/daemon_unix.go]: right, ty!

**Comment by duglin** [comment]
LGTM once janky is done

**Comment by cpuguy83** [comment]
LGTM

**Pull request merged**